### PR TITLE
Feature/rw hook up timer endpoints

### LINKED ISSUE #43: Hook up to BE endpoint to get Timer Settings
### Feature Categories:

- React 

**Describe the feature/Bug your gonna work on**
- In Timer view, add useEffect hook to get settings from BE on load

### LINKED ISSUE #45: Hook up to BE endpoint to update settings
### Feature Categories:

- React  

**Describe the feature/Bug your gonna work on**

- On Settings component, connect to BE endpoint to update work/break intervals when user selects new ones

### PR DESCRIPTION
**Pull request by rwilliams659**
# Som Timer Pull Request Form

## Description
- We hooked the app up to the BE to:
    - Get work and break interval from settings on page load (within our settingsContext) 
    - Update settings in back end after a user updates them on the settings page 
- Timer files updated to be named "CountdownTimer", which is why the changes look like so much - this really just reflects the name update!
## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other, Please explain:

**We have discovered a bug with this process, where the React Compound Timer component does not get its state updated immediately with the updated settings times, and thus on load the info is incorrect until you navigate to a different page & come back to the timer**

## Has This Been Tested?
- [ ] No
- [X] Yes
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration:
`npm test`
# Message for reviewer:
`Your message here`
# Checklist:
- [X] My code has no unused/commented out code
- [X] My code has no console logs
- [X] I have performed a self-review of my own code
- [X] I have commented on my code, particularly in hard-to-understand areas

resolve #43 
resolve #45

### CODE REVIEW

File: src/apiCalls.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import axios from 'axios'
+
+export const getSettings = () => {
+  return axios.get('https://som-timer-be.herokuapp.com/api/timers/1')
+}
+
+export const updateSettings = (settings) => {
```
Comment (jkwest-93) [inline src/apiCalls.js:7]: This looks great, y'all! Nice work with this network request.

**Review by jkwest-93 — APPROVED** [review]

File: src/Components/CountdownTimer/CountdownTimer.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+            time: settings.workInterval * 30000,
+            callback: () => countdownColorChange("#f06b64"),
+          },
+          {
+            time: settings.workInterval * 15000,
+            callback: () => countdownColorChange("#e44f47"),
+          },
+        ]}
+      >
+        {({ start, pause, reset, getTime}) => (
```
Comment (jkwest-93) [inline src/Components/CountdownTimer/CountdownTimer.js:53]: Such a bummer that we couldn't get the response to work with our visual countdown functionality... I'm confident we can figure out a way to get this working though!

File: src/Components/Settings/Settings.js [hunk 3]
```diff
@@ -23,7 +28,7 @@ const Settings = () => {
           max={240}
           id='workInterval'
           placeholder={settings.workInterval}
-          onChange={onChange}
+          onBlur={onChange}
```
Comment (Abdeboskey) [inline src/Components/Settings/Settings.js:31]: ''Twas fun figuring this out!

File: src/apiCalls.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import axios from 'axios'
```
Comment (Abdeboskey) [inline src/apiCalls.js:1]: I'm really glad that we followed through with using axios to get that experience!

**Review by Abdeboskey — APPROVED** [review]
Looking forward to squashing this weird bug, but otherwise great teamwork on this today y'all. ''Twas really fun!

File: src/Components/CountdownTimer/CountdownTimer.js [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+              </button>
+            </div>
+          </>
+        )}
+      </Timer>
+    </div>
+  );
+}
+
+export default CountdownTimer;
```
Comment (sienna-kopf) [inline src/Components/CountdownTimer/CountdownTimer.js:136]: I would love a walkthrough of what is going on here at some point!

File: src/apiCalls.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+
+export const getSettings = () => {
+  return axios.get('https://som-timer-be.herokuapp.com/api/timers/1')
```
Comment (sienna-kopf) [inline src/apiCalls.js:4]: so sweet how this call to the BE works! I love seeing how it all fits together!

**Review by sienna-kopf — APPROVED** [review]

**Pull request merged**